src/KustoExpressionParser.ts

```diff
@@ -43,17 +43,25 @@ export class KustoExpressionParser {
     };
 
     const parts: string[] = [];
+    const isDynamic = query.search.property.name.includes(DYNAMIC_TYPE_ARRAY_DELIMITER);
```

Comment:
I know the schema has a property `isDynamic` that we could check instead but checking this way avoids having to filter through that list.

Reply:
the delimiter is `'["`indexer`"]'` so safe to say it will not be included in the property name by chance 👍

Reply:
nit: I would call this `isDynamicArray` to be more accurate